File: 1107-minimum-swaps-to-group-all-1s-together/minimum-swaps-to-group-all-1s-together.java
class Solution {
    public int minSwaps(int[] data) {
        int count = 0;
        for (int i = 0; i < data.length; i++) {
            if (data[i] == 1) count++;
        }

        int countZeros = 0;
        int minZeros = Integer.MAX_VALUE;
        for (int i = 0; i < count; i++) {
            if (data[i] == 0) countZeros++;
        }
        minZeros = Math.min(minZeros, countZeros);
        int left = 0;
        for (int i = count; i < data.length; i++) {
            if (data[i] == 0) {
                countZeros++;
            }
            if (data[left] == 0) {
                countZeros--;
            }
            minZeros = Math.min(minZeros, countZeros);
            left++;
        }
        return minZeros;
    }
}

File: 1107-minimum-swaps-to-group-all-1s-together/minimum-swaps-to-group-all-1s-together.py
class Solution:
    def minSwaps(self, data: List[int]) -> int:
        count = data.count(1)
        zeroes = 0
        for i in range(count):
            if data[i] == 0:
                zeroes += 1
        minSwaps = zeroes
        for i in range(count, len(data)):
            if data[i - count] == 0:
                zeroes -= 1
            if data[i] == 0:
                zeroes += 1
            minSwaps = min(minSwaps, zeroes)
        return minSwaps